<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>XVideoUIClass</class>
 <widget class="QWidget" name="XVideoUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1204</width>
    <height>815</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>XVideoUI</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#XVideoUIClass
{
background-color: rgb(53, 53, 53);
}

#closeButton
{
font: 75 11pt &quot;等线&quot;;
color: rgb(255, 255, 255);
background-color: rgb(53, 53, 53);
}
QPushButton:!hover
{
    background-color: qlineargradient(spread:pad,x1:0,y1:0,x2:0,y2:1,stop:0 rgba(120, 120, 120, 255),stop:0.495      rgba(39, 39, 39, 255),stop:0.505 rgba(39, 39, 39, 255),stop:1 rgba(120, 120, 120, 255));
    border:none;
    border-radius:5px;
	color: rgb(255, 255, 255);
	font: 75 12pt &quot;黑体&quot;;
}
QPushButton:hover
{
    background-color: qlineargradient(spread:pad,x1:0,y1:0,x2:0,y2:1,stop:0 rgba(39, 39, 39, 255),stop:0.495      rgba(120, 120, 120, 255),stop:0.505 rgba(120, 120, 120, 255),stop:1 rgba(39, 39, 39, 255));
    border:none;
    border-radius:5px;
	color: rgb(255, 255, 255);
	font: 75 12pt &quot;黑体&quot;;
}
QLabel
{
font: 75 12pt &quot;黑体&quot;;
color: rgb(255, 255, 255);
}
#playButton{background-color: rgba(255, 255, 255,0);}

#playButton:hover{image: url(:/XVideoUI/Resources/PlayN.png);}

#playButton:!hover{image: url(:/XVideoUI/Resources/PlayH.png);}

#playButton:pressed{image: url(:/XVideoUI/Resources/PlayP.png);}

#pauseButton{background-color: rgba(255, 255, 255,0);}

#pauseButton:hover{image: url(:/XVideoUI/Resources/PauseN.png);}

#pauseButton:!hover{image: url(:/XVideoUI/Resources/PauseH.png);}

#pauseButton:pressed{image: url(:/XVideoUI/Resources/PauseP.png);}</string>
  </property>
  <widget class="QPushButton" name="closeButton">
   <property name="geometry">
    <rect>
     <x>1170</x>
     <y>10</y>
     <width>31</width>
     <height>21</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string>X</string>
   </property>
   <property name="flat">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="XVideoWidget" name="src1Video">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>560</width>
     <height>452</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="openButton">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>490</y>
     <width>93</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
  </widget>
  <widget class="QSlider" name="playSlider">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>500</y>
     <width>561</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>600</x>
     <y>550</y>
     <width>121</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>亮度[0-100]</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>560</x>
     <y>580</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>对比度[1.0-3.0]</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="bright">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>550</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="contrast">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>580</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
   <property name="value">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="setButton">
   <property name="geometry">
    <rect>
     <x>940</x>
     <y>550</y>
     <width>93</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>设置</string>
   </property>
  </widget>
  <widget class="XVideoWidget" name="des">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>20</y>
     <width>560</width>
     <height>452</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="exportButton">
   <property name="geometry">
    <rect>
     <x>760</x>
     <y>490</y>
     <width>93</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>导出</string>
   </property>
  </widget>
  <widget class="QPushButton" name="playButton">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>520</y>
     <width>51</width>
     <height>51</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="pauseButton">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>520</y>
     <width>51</width>
     <height>51</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>610</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图像旋转</string>
   </property>
  </widget>
  <widget class="QComboBox" name="rotate">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>610</y>
     <width>87</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>0</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>90</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>180</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>270</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>650</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图像镜像</string>
   </property>
  </widget>
  <widget class="QComboBox" name="flip">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>650</y>
     <width>87</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>不处理</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>左右镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下左右镜像</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>700</y>
     <width>121</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图像尺寸W,H</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="width">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>700</y>
     <width>121</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="height">
   <property name="geometry">
    <rect>
     <x>850</x>
     <y>700</y>
     <width>121</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="pyup">
   <property name="geometry">
    <rect>
     <x>850</x>
     <y>740</y>
     <width>121</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="value">
    <number>0</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="pydown">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>740</y>
     <width>121</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>740</y>
     <width>271</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图像金字塔（高斯，拉普拉斯）</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>780</y>
     <width>271</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图像裁剪(x,y,width,height)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="cx">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>780</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="cy">
   <property name="geometry">
    <rect>
     <x>790</x>
     <y>780</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="cw">
   <property name="geometry">
    <rect>
     <x>850</x>
     <y>780</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="ch">
   <property name="geometry">
    <rect>
     <x>910</x>
     <y>780</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QComboBox" name="color">
   <property name="geometry">
    <rect>
     <x>930</x>
     <y>610</y>
     <width>87</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>RGB图像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>灰度图</string>
    </property>
   </item>
  </widget>
  <widget class="QPushButton" name="markButton">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>550</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>水印(x,y,a)</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="mx">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>570</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
   <property name="value">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="my">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>570</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
   <property name="value">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ma">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>570</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
   <property name="value">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="XVideoWidget" name="src2Video">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>600</y>
     <width>236</width>
     <height>212</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="blendButton">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>600</y>
     <width>93</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>融合</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ba">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>600</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
   <property name="value">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="merge">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>650</y>
     <width>93</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>合并</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>XVideoWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>xvideowidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="XVideoUI.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1170</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>633</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>630</x>
     <y>490</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderPressed()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>sliderPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>570</x>
     <y>502</y>
    </hint>
    <hint type="destinationlabel">
     <x>588</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>sliderRelease()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>460</x>
     <y>507</y>
    </hint>
    <hint type="destinationlabel">
     <x>486</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>setPos(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>570</x>
     <y>509</y>
    </hint>
    <hint type="destinationlabel">
     <x>655</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>set()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>982</x>
     <y>560</y>
    </hint>
    <hint type="destinationlabel">
     <x>960</x>
     <y>-27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exportButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>exportVideo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>801</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>825</x>
     <y>-81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>260</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>-18</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>markButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>mark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>568</y>
    </hint>
    <hint type="destinationlabel">
     <x>172</x>
     <y>-27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blendButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>blend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>612</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>-19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>merge</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>merge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>315</x>
     <y>666</y>
    </hint>
    <hint type="destinationlabel">
     <x>347</x>
     <y>-27</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>set()</slot>
  <slot>sliderPress()</slot>
  <slot>sliderRelease()</slot>
  <slot>setPos(int)</slot>
  <slot>exportVideo()</slot>
  <slot>play()</slot>
  <slot>pause()</slot>
  <slot>mark()</slot>
  <slot>blend()</slot>
  <slot>merge()</slot>
 </slots>
</ui>
